controllers/drplacementcontrol_controller.go

```diff
@@ -637,7 +637,13 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 	if isBeingDeleted(drpc, placementObj) {
 		// DPRC depends on User PlacementRule/Placement. If DRPC or/and the User PlacementRule is deleted,
 		// then the DRPC should be deleted as well. The least we should do here is to clean up DPRC.
-		return r.processDeletion(ctx, drpc, placementObj, logger)
+		err := r.processDeletion(ctx, drpc, placementObj, logger)
+		if err != nil {
+			// update drpc progression only on err
+			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
+			return r.setProgressionToDelete(drpc, logger)
```

Comment:
Maybe change the function to: `setProgressionAndUpdate`
That way, you can reuse the function `setProgression` and then just call update if the progression has changed. What I am trying to say is this:
```
setProgressionAndUpdate(ProgressionSetDeleting)
...
...
func (r *DRPlacementControlReconciler) setProgressionAndUpdate(...) {
    setProgression(...)
    Update(...)
}
```